src/Geocoder/Provider/What3wordsProvider.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Geocoder\Provider;
+
+use Geocoder\Exception\InvalidCredentials as InvalidCredentialsException;
+use Geocoder\Exception\NoResult as NoResultException;
+use Geocoder\Exception\UnsupportedOperation as UnsupportedException;
+
+use Ivory\HttpAdapter\HttpAdapterInterface;
+
+class What3wordsProvider extends AbstractHttpProvider implements Provider
+{
+    const FORWARD_URL = 'https://api.what3words.com/v2/forward?addr=%s&key=%s';
+    const REVERSE_URL = 'https://api.what3words.com/v2/reverse?coords=%F,%F&key=%s';
+
+    private $apiKey = null;
+
+    public function __construct(HttpAdapterInterface $adapter, $apiKey)
+    {
+        parent::__construct($adapter);
+
+        $this->apiKey = $apiKey;
+    }
+
+    public function getName()
+    {
+        return 'what3words';
+    }
+
+    public function geocode($value)
+    {
+        if (filter_var($value, FILTER_VALIDATE_IP)) {
+            throw new UnsupportedException('The what3words provider does not support IP addresses.');
+        }
+
+        if (null == $this->apiKey) {
+            throw new InvalidCredentialsException('No what3words API key provided.');
+        }
+
+        $query = sprintf(self::FORWARD_URL, urlencode($value), $this->apiKey);
+        return $this->executeQuery($query);
+    }
+
+    public function reverse($latitude, $longitude)
+    {
+        if (null == $this->apiKey) {
+            throw new InvalidCredentialsException('No what3word API key provided.');
+        }
+
+        $query = sprintf(self::REVERSE_URL, $latitude, $longitude, $this->apiKey);
```

Comment:
Err. Am I missing something here?

There are checks for a supplied API key for both the `geocode` method (at line 36) and `reverse` method (at line 46) ...